src/components/AnonymousChat/BaseChatItem.tsx

```diff
@@ -82,21 +79,24 @@ const BaseChatItem = ({item, index, type}: BaseChatItemComponentProps) => {
         message={item?.message}
         time={calculateTime(item?.updatedAt, true)}
         username={handleUserName(item)}
-        type={BaseChatItemTypeProps.MY_ANON_CHAT}
         status={item?.status as ChatStatus}
-        chatType={type}
+        chatType={type ?? ANONYMOUS}
+        messageType={item?.type}
+        data={item?.rawJson}
       />
     );
 
   return (
     <ChatItemTargetText
       key={index}
       avatar={handleAvatar()}
-      isContinuous={item?.isContinuous}
+      isContinuous={item?.isContinuous ?? false}
       message={item?.message}
       time={calculateTime(item?.updatedAt, true)}
       username={handleUserName(item)}
-      type={BaseChatItemTypeProps.ANON_CHAT}
+      chatType={type ?? ANONYMOUS}
+      messageType={item?.type ?? 'regular'}
+      data={item?.rawJson}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `type` prop in both `ChatItemMyText` and `ChatItemTargetText` components has been removed and replaced with `messageType={item?.type}` and `messageType={item?.type ?? 'regular'}` respectively. Ensure that the `item?.type` value aligns with the expected values for `messageType`. Also, the fallback value `'regular'` should be validated to ensure it doesn't cause any unexpected behavior.
```diff
-        type={BaseChatItemTypeProps.MY_ANON_CHAT}
+        messageType={item?.type}

-       type={BaseChatItemTypeProps.ANON_CHAT}
+       messageType={item?.type ?? 'regular'}
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->